lib/spack/spack/package.py

```diff
@@ -1420,6 +1524,43 @@ def rpath_args(self):
         return " ".join("-Wl,-rpath,%s" % p for p in self.rpath)
 
 
+class Package(PackageBase):
+    phases = ['install', 'log']
+    # This will be used as a registration decorator in user
+    # packages, if need be
+    PackageBase.sanity_check('install')(PackageBase.sanity_check_prefix)
+
+
+class AutotoolsPackage(PackageBase):
+    phases = ['autoreconf', 'configure', 'build', 'install', 'log']
+
+    def autoreconf(self, spec, prefix):
+        """Not needed usually, configure should be already there"""
+        pass
+
+    @PackageBase.sanity_check('autoreconf')
+    def is_configure_or_die(self):
+        if not os.path.exists('configure'):
+            raise RuntimeError('configure script not found in {0}'.format(os.getcwd()))
+
+    def configure_args(self):
+        return list()
```

Comment:
How would spack setup work if packages have non standardized phase names?
(This is a real question, not rhetorical.  I'm looking to better understand
the proposal.)
On Jul 8, 2016 5:03 PM, "Denis Davydov" <notifications@github.com> wrote:

> In lib/spack/spack/package.py:
>
> > +
> > +
> > +class AutotoolsPackage(PackageBase):
> > +    phases = ['autoreconf', 'configure', 'build', 'install', 'log']
> > +
> > +    def autoreconf(self, spec, prefix):
> > +        """Not needed usually, configure should be already there"""
> > +        pass
> > +
> > +    @PackageBase.sanity_check('autoreconf')
> > +    def is_configure_or_die(self):
> > +        if not os.path.exists('configure'):
> > +            raise RuntimeError('configure script not found in {0}'.format(os.getcwd()))
> > +
> > +    def configure_args(self):
> > +        return list()
>
> i agree that uniformity of arguments is good for package developers.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LLNL/spack/pull/1186/files/fa2783b9dc8c13e675a9b6791b240cdca61e4307#r70140526>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AB1cd042MNpF95q9LVhD55nuIBXUpqH-ks5qTrscgaJpZM4JG8iX>
> .
>